testing/s3mock/src/main/java/org/projectnessie/s3mock/S3Resource.java

```diff
@@ -362,6 +370,66 @@ public Response getObject(
         });
   }
 
+  @PUT
+  @Path("/{bucketName:[a-z0-9.-]+}/{object:.+}")
+  @Consumes(MediaType.WILDCARD)
+  // See https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html
+  public Response putObject(
+      @PathParam("bucketName") String bucketName,
+      @PathParam("object") String objectName,
+      @Context HttpHeaders headers,
+      InputStream stream
+      //    @HeaderParam(RANGE) Range range,
+      //    @HeaderParam(IF_MATCH) List<String> match,
+      //    @HeaderParam(IF_NONE_MATCH) List<String> noneMatch,
+      //    @HeaderParam(IF_MODIFIED_SINCE) Date modifiedSince,
+      //    @HeaderParam(IF_UNMODIFIED_SINCE) Date unmodifiedSince
+      ) {
+    return withBucket(
+        bucketName,
+        bucket -> {
+          byte[] data;
+
+          String contentEncoding = headers.getHeaderString("Content-Encoding");
+          try {
+            data = chunkedInput(contentEncoding, stream).readAllBytes();
+          } catch (Exception e) {
+            return Response.status(500, e.toString()).build();
+          }
+
+          byte[] md5 = Hashing.md5().hashBytes(data).asBytes();
+          String md5str = Base64.getEncoder().encodeToString(md5);
+          String contentMD5 = headers.getHeaderString("Content-MD5");
+          String contentType = headers.getHeaderString("Content-Type");
+          if (contentMD5 != null && !contentMD5.equals(md5str)) {
+            return Response.status(400, "Content-MD5 does not match content").build();
+          }
+          try {
+            bucket.putObject().putObject(objectName, contentType, data);
+            return Response.ok()
+                // .header("ETag", asQuotedHex(md5))
+                .header("Date", RFC_1123_DATE_TIME.format(Instant.now().atZone(ZoneId.of("UTC"))))
+                .build();
+          } catch (UnsupportedOperationException e) {
+            return Response.status(405, "PUT object not allowed").build();
+          }
+        });
+  }
+
+  private InputStream chunkedInput(String contentEncoding, InputStream input) throws IOException {
+    if (contentEncoding != null) {
+      List<String> contentEncodingList =
+          Arrays.stream(contentEncoding.split(",")).collect(Collectors.toList());
+      if (contentEncodingList.remove("identity")) {
```

Comment:
`Collectors.toList()` offers no mutability guarantee. Calling `remove` on the returned list seems risky.